src/segment-loader.js

```diff
@@ -1284,6 +1284,17 @@ export default class SegmentLoader extends videojs.EventTarget {
 
     if (segmentInfo.timestampOffset !== null &&
         segmentInfo.timestampOffset !== this.sourceUpdater_.timestampOffset()) {
+      const buffered = this.buffered_();
+
+      // Update the timestampOffset with more accurate timing info from the
+      // current buffered range and results of the probe, if it is available
+      if (buffered.length && timingInfo && timingInfo.segmentTimestampInfo) {
+        const ptsStartTime = timingInfo.segmentTimestampInfo[0].ptsTime;
+        const dtsStartTime = timingInfo.segmentTimestampInfo[0].dtsTime;
+
+        segmentInfo.timestampOffset = buffered.end(buffered.length - 1) - (ptsStartTime - dtsStartTime);
```

Comment:
Instead of updating `timestampOffset` here and using `buffered.end` I think we should fix the source of the issue, i.e. `segment.end` and then just shift the `timestampOffset` by pts/dts difference here.

>Currently, the value for timestampOffset comes from segment.end, which is set from the DTS values returned by the tsprobe. This is not the correct value to use for the timestampOffset

I wouldn't say its not the correct value, we want to set the `timestampOffset` to the end of the previous segment so the next segment gets placed immediately after it, its just "incorrect" because `segment.end` is given the wrong value.